pkg/transform/oauth/github.go

```diff
@@ -12,18 +12,27 @@ import (
 //IdentityProviderGitHub is a Github specific identity provider
 type IdentityProviderGitHub struct {
 	identityProviderCommon `yaml:",inline"`
-	GitHub                 struct {
-		HostName string `yaml:"hostname"`
-		CA       struct {
-			Name string `yaml:"name"`
-		} `yaml:"ca"`
-		ClientID     string `yaml:"clientID"`
-		ClientSecret struct {
-			Name string `yaml:"name"`
-		} `yaml:"clientSecret"`
-		Organizations []string `yaml:"organizations"`
-		Teams         []string `yaml:"teams"`
-	} `yaml:"github"`
+	GitHub                 GitHub `yaml:"github"`
+}
+
+// GitHub provider specific data
+type GitHub struct {
+	HostName      string             `yaml:"hostname"`
+	CA            GitHubCA           `yaml:"ca"`
+	ClientID      string             `yaml:"clientID"`
+	ClientSecret  GitHubClientSecret `yaml:"clientSecret"`
+	Organizations []string           `yaml:"organizations"`
+	Teams         []string           `yaml:"teams"`
+}
+
+//GitHubCA is a CA for a GitHub provider
+type GitHubCA struct {
+	Name string `yaml:"name"`
+}
+
+//GitHubClientSecret is a client secret for GitHub
+type GitHubClientSecret struct {
+	Name string `yaml:"name"`
```

Comment:
I would move `GitHubClientSecret`, `GitlabClientSecret`, and `GoogleClientSecret` to `types.go` and rename it as `ClientSecret`.

```golang
type ClientSecret struct {
    Name string `yaml:"name"`
}
```